### PR TITLE
convert several js files to ts

### PR DESCRIPTION
**Pull request by DevChu**

### CODE REVIEW

File: zk/src/archive/web/js/zk/fmt/numfmt.ts [hunk 1]
```diff
@@ -41,20 +41,20 @@ it will be useful, but WITHOUT ANY WARRANTY.
 			val = valStr.substring(0, j) + val;
 		return k ? '1' + val : val;
 	}
-	function compareHalf(valStr, ri) {
+	function compareHalf(valStr, ri): number {
 		var result,
-			base = '5';
```
Comment (scribetw) [inline zk/src/archive/web/js/zk/fmt/numfmt.ts:46]: '5'.toCharCode() ?

**Review by scribetw — CHANGES_REQUESTED** [review]
numfmt viewed, start from gapi

File: zk/src/archive/web/js/zk/mount.ts [hunk 2]
```diff
@@ -476,15 +493,15 @@ function zkamn(pkg, fn) {
 			if (wi) {
 				if (wi[0] === 0) { //page
 					var props = wi[2];
-					zkdt(zk.cut(props, 'dt'), zk.cut(props, 'cu'), zk.cut(props, 'uu')
-						, zk.cut(props, 'rsu'), zk.cut(props, 'ru'));
+					zkdt(zk.cut(props, 'dt').toString(), zk.cut(props, 'cu').toString(), zk.cut(props, 'uu').toString()
```
Comment (scribetw) [inline zk/src/archive/web/js/zk/mount.ts:496]: check undefined

File: zk/src/archive/web/js/zk/mount.ts [hunk 3]
```diff
@@ -528,27 +547,29 @@ function zkamn(pkg, fn) {
...
 	},
 	// register data-attributres handler (since 8.0.0
 	zkdh: function (name, script) {
 		zk.addDataHandler('data-' + name, script);
 	}
   });
-
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
 })(window);
```
Comment (scribetw) [inline zk/src/archive/web/js/zk/mount.ts:566]: remove window

File: zk/src/archive/web/js/zk/mount.ts [hunk 4]
```diff
@@ -722,14 +743,14 @@ jq(function () {
...
-			if (zk.ie < 11)
+			if (zk.ie && zk.ie < 11)
 				evt.which = 3;
 			var wevt = new zk.Event(wgt, 'onRightClick', evt.mouseData(), {}, evt);
 			_doEvt(wevt);
 			if (wevt.domStopped)
 				return false;
 		}
-		return !(zk.ie < 11) || evt.returnValue;
+		return !(zk.ie && zk.ie < 11) || evt['returnValue'];
```
Comment (scribetw) [inline zk/src/archive/web/js/zk/mount.ts:753]: evt: Event

File: zk/src/archive/web/js/zk/widget.ts [hunk 5]
```diff
@@ -4667,7 +4667,7 @@ _doFooSelect: function (evt) {
 	 */
 	fromPageCoord: function (x, y) {
 		var ofs = zk(this).revisedOffset();
-		return [x - ofs[0], y - ofs[1]];
+		return [ofs[0] ? x - ofs[0] : x, ofs[1] ? y - ofs[1] : y];
```
Comment (scribetw) [inline zk/src/archive/web/js/zk/widget.ts:4670]: no change

**Review by scribetw — CHANGES_REQUESTED** [review]

File: zk/src/archive/web/js/zk/mount.ts (hunk 4 above)
Comment (DevChu) [reply zk/src/archive/web/js/zk/mount.ts:753]: jqEvent no returnValue

File: @types/zk/zFlex.d.ts [hunk 6]
```diff
@@ -23,9 +23,9 @@ declare namespace zk {
 		beforeSize(ctl, opts, cleanup: boolean): void;
 		beforeSizeClearCachedSize(ctl, opts, cleanup: boolean): void;
 		beforeSizeForRead(): void;
-		clearCSSFlex(wgt: zk.Widget, o: FlexOrient, clearAllSiblings?: boolean): void;
+		clearCSSFlex(wgt: zk.Widget, o: string, clearAllSiblings?: boolean): void;
```
Comment (DevChu) [inline @types/zk/zFlex.d.ts:26]: FlexOrient

File: zk/src/archive/web/js/zk/js.ts [hunk 7]
```diff
@@ -1,5 +1,6 @@
 /* eslint-disable one-var */
-/* js.js
+// @ts-nocheck
```
Comment (DevChu) [inline zk/src/archive/web/js/zk/js.ts:2]: remove

File: zk/src/archive/web/js/zk/mount.ts [hunk 8]
```diff
@@ -89,16 +98,18 @@ function zkamn(pkg, fn) {
...
 		}
-		_paci.i = setInterval(function () {
+		if (!_paci) return;
+		_paci['i'] = setInterval(function () {
 			var stateless;
-			if ((zk.booted && !zk.mounting) || (stateless = _stateless()))
+			if (_paci != null && ((zk.booted && !zk.mounting) || (stateless = _stateless())))
 				if (stateless || _paci.s == _paci.e) { //done
-					clearInterval(_paci.i);
+					clearInterval(_paci['i']);
```
Comment (DevChu) [inline zk/src/archive/web/js/zk/mount.ts:112]: interface add i: number | undefined

File: zk/src/archive/web/js/zk/mount.ts [hunk 9]
```diff
@@ -475,16 +499,21 @@ function zkamn(pkg, fn) {
...
-					var props = wi[2];
-					zkdt(zk.cut(props, 'dt'), zk.cut(props, 'cu'), zk.cut(props, 'uu')
-						, zk.cut(props, 'rsu'), zk.cut(props, 'ru'));
+					var props = wi[2],
+						dtid = zk.cut(props, 'dt'),
+						cu = zk.cut(props, 'cu'),
+						uu = zk.cut(props, 'uu'),
+						rsu = zk.cut(props, 'rsu'),
+						ru = zk.cut(props, 'ru');
+					zkdt(dtid ? dtid.toString() : undefined, cu ? cu.toString() : undefined, uu ? uu.toString() : undefined,
```
Comment (DevChu) [inline zk/src/archive/web/js/zk/mount.ts:508]: zk.cut(props, 'dt') as string | undefined

**Review by scribetw — APPROVED** [review]

**Review by plijyfes — APPROVED** [review]

**Pull request merged**